Data_Engine/Modify/GroupByName.cs

```diff
@@ -0,0 +1,80 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2018, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Data.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BH.Engine.Data
+{
+    public static partial class Modify
+    {
+        /*************************************/
+        /**** Public Methods              ****/
+        /*************************************/
+
+        public static Tree<T> GroupByName<T>(this Tree<T> tree)
```

Comment:
This method has been migrated from the UI: https://github.com/BHoM/BHoM_UI/blob/2ad2b8a3a7c1a27596a5c88097da34d28636e8f8/UI_Engine/Modify/GroupMethodsByName.cs#L39
Old:
```
public static Tree<MethodBase> GroupMethodsByName<MethodBase>(this Tree<MethodBase> tree)
```
New:
```
public static Tree<T> GroupByName<T>(this Tree<T> tree)
```

Note that the name `MethodBase` was actually used as the name of a generic parameter, and was not a real specialisation of a generic method.